src/Uno.UI.DualScreen/DuoApplicationViewSpanningRect.cs

```diff
@@ -31,91 +32,114 @@ public IReadOnlyList<Rect> GetSpanningRects()
 			{
 				InitializeHelper(currentActivity);
 
-				if (_isDualScreenDevice.Value && _helper.IsDualMode)
+				if (_isDualScreenDevice.Value)
 				{
-					var wuxWindowBounds = ApplicationView.GetForCurrentView().VisibleBounds.LogicalToPhysicalPixels();
-					var helperOrientation = _helper.GetRotation();
-					var wuOrientation = DisplayInformation.GetForCurrentView().CurrentOrientation;
+					if (_helper.IsDualMode)
+					{
+						var helperOrientation = _helper.GetRotation();
+						if (_previousMode.orientation == helperOrientation && _previousMode.result != null)
+						{
+							return _previousMode.result;
+						}
 
-					var occludedRects = _helper
-						.DisplayMask
-						.GetBoundingRectsForRotation(helperOrientation)
-						.Select(r => (Rect)r)
-						.ToArray();
+						_previousMode.orientation = helperOrientation;
+						_previousMode.result = null;
 
-					if (occludedRects.Length > 0)
-					{
-						Rect bounds = wuxWindowBounds;
-						var occludedRect = occludedRects[0];
-						var intersection = bounds;
-						intersection.Intersect(occludedRect);
+						var wuxWindowBounds = ApplicationView.GetForCurrentView().VisibleBounds.LogicalToPhysicalPixels();
+						var wuOrientation = DisplayInformation.GetForCurrentView().CurrentOrientation;
 
-						if (wuOrientation == DisplayOrientations.Portrait || wuOrientation == DisplayOrientations.PortraitFlipped)
+						var occludedRects = _helper
+							.DisplayMask
+							.GetBoundingRectsForRotation(helperOrientation)
+							.SelectToArray(r => (Rect)r); // convert to managed Rect
+
+						if (occludedRects.Length > 0)
 						{
-							// Compensate for the status bar size (the occluded area is rooted on the screen size, whereas
-							// wuxWindowBoundsis rooted on the visible size of the window, unless the status bar is translucent.
-							if (bounds.X == 0 && bounds.Y == 0)
+							if (occludedRects.Length > 1 && this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Warning))
 							{
-								var statusBarRect = StatusBar.GetForCurrentView().OccludedRect.LogicalToPhysicalPixels();
-								occludedRect.Y -= statusBarRect.Height;
+								this.Log().Warn($"DualMode: Unknown screen layout, more than one occluded region. Only first will be considered. Please report your device to Uno Platform!");
 							}
-						}
 
-						if (intersection != Rect.Empty)
-						{
-							if (occludedRect.X == bounds.X)
+							var bounds = wuxWindowBounds;
+							var occludedRect = occludedRects[0];
+							var intersection = bounds;
+							intersection.Intersect(occludedRect);
+
+							//if (wuOrientation == DisplayOrientations.Portrait || wuOrientation == DisplayOrientations.PortraitFlipped)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4993239)